packages/grafana-ui/src/components/Graph/GraphContextMenu.tsx

```diff
@@ -80,8 +83,26 @@ export const GraphContextMenu: React.FC<GraphContextMenuProps> = ({
       />
     );
   };
+  const renderMenuGroupItems = () => {
```

Comment:
This function is used in a few places, it should be abstracted to function that takes `itemsToRender` and renders them.

Reply:
Good point. But I was wondering: if we should abstract this function into a function that takes `itemsToRender`, won't it lead it to the function being called when passed as a value via a HOC? @Clarity-89

Reply:
Not sure I understand what you mean. What I was suggesting is to create a global function, or even better a component that accepts `itemsToRender`, so instead of creating `renderMenuGroupItems` function inside each component, we can just call the, lets call it `MenuItems`, e.g. `<MenuItems itemsToRender={itemsGroup} />`.

Reply:
Isn't that what we're trying to eradicate? @dprokop

Reply:
Well, yes and no. I see @Clarity-89 point here, but I don't think we need to do this. The menu component is fairly internal, used for context menu primarily and I think the refactor we are doing here has pure context menu creation. 

The goal of the work in the pr is to make menu items rendering independent from the context menu implementation.

Reply:
Alright.
Thanks @dprokop and @Clarity-89 for your great suggestions! Let's leave it as it is currently since this function isn't really used in too many places.